lib/board.rb

```diff
@@ -0,0 +1,62 @@
+class Board
+
+  attr_accessor :cells
+
+  def initialize
+    self.cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+  end
+
+  def reset!
+    @cells = Array.new(9, " ")
+    @cells
+  end
+
+def display
+    puts " "
+    puts "     Here is the current board:"
+    puts " "
+    puts "      #{self.cells[0]} | #{self.cells[1]} | #{self.cells[2]} "
+    puts "     -----------"
+    puts "      #{self.cells[3]} | #{self.cells[4]} | #{self.cells[5]} "
+    puts "     -----------"
+    puts "      #{self.cells[6]} | #{self.cells[7]} | #{self.cells[8]} "
+  end
+
+
+  def position(number)
+    self.cells[number.to_i - 1]
+  end
+
+  def full?
+    return true if self.cells.none? { |position| position == " " }
+  end
+
+  def turn_count
+  self.cells.select {|position| position != " "}.count
+  end
+
+  def taken?(number)
+    if self.position(number) == "X" || self.position(number) == "O"
+      return true
+      else false
+    end
+  end
+
+  def valid_move?(number)
+    number = number.to_i
+    if number < 1 || number > 9
```

Comment:
You can simplify checking the valid integer range, and make it more semantic, by using [`Comparable#between?`](http://ruby-doc.org/core-2.2.3/Comparable.html#method-i-between-3F).
`number.between?(1,9)`